### DISCUSSION TITLE
only install dependencies of a binary

### DISCUSSION THREAD

**Discussion by Geertsky** [dracutdevs/dracut#2657] [Q&A]
Hello all!
I'm writing a module. Now, when I install a binary using `inst <binary> [<optional destination>]`, the binary gets installed. Including the dependencies (libraries) of the binary. 
I found out, this library dependencies resolution step doesn't happen at the same moment as the installation of the binary. I guess that the image first gets populated with all the files/binaries that modules install in one way or the other, and at the end, just before the image gets packed, there is a loop over all the binaries in the image which installs all the library dependencies.
Would it be possible to only install the dependencies of a binary but not the binary itself?
As an example:
```
ldd /bin/python
        linux-vdso.so.1 (0x00007f8e5b10f000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f8e5b0eb000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f8e5b0e6000)
        libutil.so.1 => /lib64/libutil.so.1 (0x00007f8e5b0e1000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f8e5a91d000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f8e5a730000)
        librt.so.1 => /lib64/librt.so.1 (0x00007f8e5b0da000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f8e5b111000)
```

Then `inst_deps_only /bin/python` would install:
```
	/lib64/libpthread.so.0
	/lib64/libdl.so.2
	/lib64/libutil.so.1
	/lib64/libm.so.6
	/lib64/libc.so.6
	/lib64/librt.so.1
```

**Comment by aafeijoo-suse** [comment]
It could be implemented, but... I'm curious, what's your use case?

**Reply by Geertsky** [reply]
I don't think it makes sense for normal dracut use cases. I'm using dracut to build an initramfs with a conda python environment included for ansible usage. The idea is to:
1) halt the execution at pre-mount
2)  partition a disk 
3) perform a minimal install
4) continue execution or reboot for selinux.

Currently I extract the python binary, place it in initramfs at build time to get the dependencies resolved, and then when at pre-mount stage I remove the binary and unpack the conda environment.

It works, but gives a bad taste. Where the, to be considered ugly "hacks", are pretty much down to none. 
Well, obviously dracut is not thought to include complete python environments, but for this use case it makes sense as it makes servers 100% parameterized.

Anyways, I thought there might already be such a "resolve dependencies" function available. Could you point me to where this resolution actually happens?

Ooohw, https://github.com/Geertsky/dracut-bambini and https://github.com/Geertsky/bambini when you're interested.

**Reply by aafeijoo-suse** [reply]
> Anyways, I thought there might already be such a "resolve dependencies" function available. Could you point me to where this resolution actually happens?

https://github.com/dracutdevs/dracut/blob/5d2bda46f4e75e85445ee4d3bd3f68bf966287b9/src/install/dracut-install.c#L510
